tips/TIP-0018/tip-0018.md

```diff
@@ -0,0 +1,3755 @@
+- For every non-zero explicit `Alias ID` on the output side there must be a corresponding alias on the input side. The
+  corresponding alias has the explicit or implicit `Alias ID` equal to that of the alias on the output side.
+
+#### Consumed Outputs
+
+Whenever an alias output is consumed in a transaction, it means that the alias is transitioned into its next state. The
+**current state** is defined as the **consumed alias output**, while the **next state** is defined as the **alias
+output with the same explicit `AliasID` on the output side**. There are two types of transitions: `state transition`
+and `governance transition`.
+- State transition:
+    - A state transition is identified by an incremented `State Index`.
+    - The `State Index` must be incremented by 1.
+    - The unlock block must correspond to the `Address` of <i>State Controller Address Unlock Condition</i>.
+    - State transition can only change the following fields in the next state: `IOTA Amount`, `Native Tokens`,
+      `State Index`, `State Metadata Length`, `State Metadata`, `Foundry Counter` and `Sender Block`.
+    - `Foundry Counter` field must increase by the number of foundry outputs created in the transaction that map to
+      `Alias ID`. The `Serial Number` fields of the created foundries must be the set of natural numbers that cover the
+       open-ended interval between the previous and next values of the `Foundry Counter` field in the alias output.
+    - The created foundry outputs must be sorted in the list of outputs by their `Serial Number`. Note, that any
+      foundry that maps to `Alias ID` and has a `Serial Number` that is less or equal to the `Foundry Counter` of the
+      input alias is ignored when it comes to sorting.
+    - Newly created foundries in the transaction that map to different aliases can be interleaved when it comes to
+      sorting.
+- Governance transition:
+    - A governance transition is identified by an unchanged `State Index` in next state. If there is no alias output on
+      the output side with a corresponding explicit `Alias ID`, the alias is being destroyed. The next state is the
+      empty state.
+    - The unlock block must correspond to the `Address` of <i>Governor Address Unlock Condition</i>.
+    - Governance transition must only change the following fields:
+      `Address` of <i>State Controller Address Unlock Condition</i>,`Address` of <i>Governor Address Unlock Condition</i>,
+      `Metadata Block`, `Sender Block`.
+    - The `Metadata Block` is optional, the governor can put additional info about the chain here, for example chain
+      name, fee structure, supported VMs, list of access nodes, etc., anything that helps clients to fetch info (i.e.
+      account balances) about the layer 2 network.
+- When a consumed alias output has an `Issuer Block` and a corresponding alias output on the output side,
+  `Issuer Block` is not allowed to change.
+
+#### Created Outputs
```

Comment:
The creation constraints of an AliasOutput (zeroed AliasId, state index, foundry counter...) are not very clear/explicit